src/lib.rs

```diff
@@ -1,7 +1,10 @@
 use std::error::Error;
 use std::io;
 
-fn get_user_input(prompt: &str) -> String {
+// CLI input output
+
+fn take_user_input(prompt: &str) -> String {
```

Comment:
Out of scope for this pr: I think such "section" comments between function definitions is a strong sign that there may be value in extracting functions to their own class / module.